File: app/src/main/java/com.github.anrimian.musicplayer/ui/common/error/ErrorCommand.java
package com.github.anrimian.musicplayer.ui.common.error;

import javax.annotation.Nonnull;

/**
 * Created on 29.10.2017.
 */

public class ErrorCommand {

    @Nonnull
    private final String message;

    public ErrorCommand(@Nonnull String message) {
        this.message = message;
    }

    @Nonnull
    public String getMessage() {
        return message;
    }
}

File: shared/app/src/main/java/com/github/anrimian/musicplayer/ui/common/error/ErrorCommand.kt
package com.github.anrimian.musicplayer.ui.common.error

/**
 * Created on 29.10.2017.
 */
open class ErrorCommand(val message: String) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ErrorCommand) return false

        if (message != other.message) return false

        return true
    }

    override fun hashCode(): Int {
        return message.hashCode()
    }

}